tuxemon/states/combat/combat_classes.py

```diff
@@ -0,0 +1,164 @@
+        ] = {}
+
+    def get(
+        self, method: Union[Technique, Condition, Item], is_flipped: bool
+    ) -> Optional[Sprite]:
+        """
+        Return a sprite usable as a method (technique, item, condition) animation.
+
+        Parameters:
+            method: Whose sprite is requested.
+            is_flipped: Flag to determine whether animation frames should be flipped.
+
+        Returns:
+            Sprite associated with the animation.
+
+        """
+        try:
+            return self._sprites[method]
+        except KeyError:
+            sprite = self.load_method_animation(method, is_flipped)
+            self._sprites[method] = sprite
+            return sprite
+
+    @staticmethod
+    def load_method_animation(
+        method: Union[Technique, Condition, Item], is_flipped: bool
+    ) -> Optional[Sprite]:
+        """
+        Return animated sprite from a technique, condition or item.
+
+        Parameters:
+            method: Whose sprite is requested.
+            is_flipped: Flag to determine whether animation frames should be flipped.
+
+        Returns:
+            Sprite associated with the animation.
+
+        """
+        if not method.animation:
+            return None
+
+        ani = AnimationEntity(method.animation)
+        if is_flipped:
+            ani.play.flip(method.flip_axes)
+        return Sprite(animation=ani.play)
+
+
+class EnqueuedAction(NamedTuple):
+    user: Union[Monster, NPC, None]
+    method: Union[Technique, Item, Condition, None]
+    target: Monster
+
+
+class ActionQueue:
+
+    def __init__(self) -> None:
+        self._action_queue: list[EnqueuedAction] = []
+
+    @property
+    def queue(self) -> list[EnqueuedAction]:
+        """Returns the current action queue."""
+        return self._action_queue
+
+    def enqueue(self, action: EnqueuedAction) -> None:
+        """Adds an action to the end of the queue."""
+        self._action_queue.append(action)
+
+    def dequeue(self, action: EnqueuedAction) -> None:
+        """Removes an action from the queue if it exists."""
+        try:
+            self._action_queue.remove(action)
+        except ValueError:
+            raise ValueError(f"Action {action} not found in queue")
+
+    def pop(self) -> EnqueuedAction:
+        """Removes and returns the last action from the queue."""
+        return self._action_queue.pop()
+
+    def is_empty(self) -> bool:
+        """Returns True if the queue is empty, False otherwise."""
+        return len(self._action_queue) == 0
+
+    def clear_queue(self) -> None:
+        """Clears the entire queue."""
+        self._action_queue.clear()
+
+    def sort(self) -> None:
+        """Sorts the queue based on the action's sort key (game rules).
+        * Techniques that damage are sorted by monster speed
+        * Items are sorted by trainer speed
+        """
+
+        def get_action_sort_key(action: EnqueuedAction) -> tuple[int, int]:
```

Comment:
Some unittests for the sorting logic would be good.

Reply:
wasted some time to implement your "would be good", do you want some more tests?

Reply:
by the way, next time "would be good" not as requested change